src/modules/workspaces/workspaces.controller.spec.ts

```diff
@@ -133,4 +133,110 @@ describe('Workspace Controller', () => {
       );
     });
   });
+
+  describe('GET /:workspaceId/members', () => {
+    const owner = newUser();
+    const workspace = newWorkspace({
+      owner,
+      attributes: { setupCompleted: true },
+    });
+
+    it('When the workspaceId is missing then it should throw BadRequestException', async () => {
+      const workspaceController = jest.spyOn(
+        workspacesController,
+        'getWorkspaceMembers',
+      );
+
+      const expectgetWorkspaceMembers = expect(async () => {
+        await workspacesController.getWorkspaceMembers(null, owner);
+      });
+      expectgetWorkspaceMembers.rejects.toThrow(BadRequestException);
+      expectgetWorkspaceMembers.rejects.toThrow('Invalid workspace ID');
+
+      expect(workspaceController).toHaveBeenCalled();
+    });
+
+    it('When the user is null or empty then it should throw UnauthorizedException', async () => {
+      const workspaceController = jest
+        .spyOn(workspacesController, 'getWorkspaceMembers')
+        .mockImplementation(async (workspaceId: string, user: any) => {
+          if (!user) throw UnauthorizedException;
+          return workspacesUsecases.getWorkspaceMembers(workspaceId, user);
+        });
```

Comment:
why are you mocking the implementation? The meaning of tests is to test the real implementation, not mocking it.  

You should mock external usecases or repositories if the controller depends on it. If the controller is too simple and you just need to add tests for coverage, then create a simple test that just checks if the controller returns what the usecase returns and that's it.